apisix/core/utils.lua

```diff
@@ -292,11 +292,16 @@ local resolve_var
 do
     local _ctx
     local n_resolved
-    local pat = [[(?<!\\)\$\{?(\w+)\}?]]
+    local pat = [[(?<!\\)\$(\{(\w+)\}|(\w+))]]
     local _escaper
 
     local function resolve(m)
-        local v = _ctx[m[1]]
+        local i = 1
+        -- if first capture's first char is "{" evaluate the second capture
+        if m[1]:byte(1) == 123 then
```

Comment:
I think you can optimise this judgement, like
```
local variable = m[2] or m [3]
local v = _ctx[variable]
```

   


```

Reply:
it works